stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -388,6 +388,143 @@ const ⋅ = dot
 const × = cross
 export ⋅, ×
 
+"""
+    hadamard(a, b)
+    a ⊙ b
+
+For arrays `a` and `b`, perform elementwise multiplication.
+`a` and `b` must have identical `axes`.
+
+`⊙` can be passed as an operator to higher-order functions.
+
+```jldoctest
+julia> a = [2, 3]; b = [5, 7];
+
+julia> a ⊙ b
+2-element Array{$Int,1}:
+ 10
+ 21
+
+julia> a ⊙ [5]
+ERROR: DimensionMismatch("Axes of `A` and `B` must match, got (Base.OneTo(2),) and (Base.OneTo(1),)")
+[...]
+```
+
+!!! compat "Julia 1.5"
+    This function requires at least Julia 1.5. In Julia 1.0-1.4 it is available from
+    the `Compat` package.
+"""
+function hadamard(A::AbstractArray, B::AbstractArray)
+    @noinline throw_dmm(axA, axB) = throw(DimensionMismatch("Axes of `A` and `B` must match, got $axA and $axB"))
+
+    axA, axB = axes(A), axes(B)
+    axA == axB || throw_dmm(axA, axB)
+    return map(*, A, B)
```

Comment:
Should this be `map(hadamard, A, B)` with `hadamard(a::Number, b::Number) = a * b`, i.e. should it be recursive?

Reply:
The closest I've ever come to wanting to do linear algebra on arrays of arrays was in representing complex numbers by 2x2 arrays. In that case, the present behaviour seems correct, and lines up with `adjoint` being recursive.

Reply:
History repeats itself...see all but the last commit and the discussion leading up to https://github.com/JuliaLang/julia/pull/35150#issuecomment-622309421.